crates/server/src/store/recursor/config.rs

```diff
@@ -110,6 +109,7 @@ fn ns_recursion_limit_default() -> u8 {
     16
 }
 
+#[cfg(feature = "dnssec")]
```

Comment:
I think the previous setup was intentional --why/how did you decide it did not make sense?

Reply:
I was flushing out conditional compilation failures and unused code warnings, I can try resolving this a different way.

Reply:
Ah, I was getting an `unreachable_pub` warning in part because the `RecursiveConfig.dnssec_policy` field is only present when DNSSEC is enabled. Since the enum's variants already change depending on DNSSEC support, I think I can safely make `RecursiveConfig.dnssec_policy` always there, which takes care of `unreachable_pub` by always using the struct definition.